lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -2580,6 +2596,14 @@ def get_column_specification(self, column, **kwargs):
             colspec += " NULL"
         return colspec
 
+    def _define_constraint_validity(self, constraint):
+        if self.dialect._supports_not_valid_constraints:
+            not_valid = constraint.dialect_options["postgresql"]["not_valid"]
+            if not_valid:
+                return " NOT VALID"
+
+        return ""
```

Comment:
it looks like NOT VALID doesn't actually change the functionality of the constraint so I guess it is safe to silently ignore for older versions; but in any case version 9.1 is super old at this point so it doesn't matter much anyway.

Reply:
our "normal support" is based at 9.6: https://docs.sqlalchemy.org/en/14/dialects/postgresql.html  so I think just render the directive if it's asked for should be fine here

Reply:
Fair enough. c2ed3e9b1c2e8510e139f40665bfcffcbf12e23d